tests/apollo/test_skvbc_preexecution.py

```diff
@@ -191,3 +192,41 @@ async def test_view_change(self, bft_network):
                 break
 
         self.assertEqual(new_last_block, last_block)
+
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_f_nonprimary_crash_amid_parallel_tx(self, bft_network):
+        '''
+        Crash f nonprimary replicas while multiple transactions are being processed.
+        Block processing of the network should be unaffected with f-count interruption.
+        Check created block lengths match submitted x count of transactions.
+        '''
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        f_count = int((len(bft_network.all_replicas()) - 1) / 3)
+        primary_index = 0
+
+        await trio.sleep(2)
+        
+        num_of_requests = 20
+        clients = bft_network.random_clients(1)
+        reader_client = list(clients)[0]
```

Comment:
I'd suggest to simply say `reader_client = bft_network.random_client()`

Reply:
Certainly. That looks much neater.